frontend/docs/patterns.md

```diff
@@ -198,6 +198,14 @@ export default PackComposerPage;
 
 ## Forms
 
+### Form buttons
+
+When building a React-controlled form, use the `Button` component for its submit button, and do not
+include a `type`, allowing it to default to `type="button"`, which is what we want. Using
+`type="submit"` leads to confusing behavior wherein the native HTML submit functionality
+interferes with our custom submit logic, causing the request to be canceled in a messy and hard to
+recognize way. Alternatively, if using `type="submit"`, ensure the submit handler calls `evt.preventDefault`
+
```

Comment:
I'd disagree with this. I think we should use button `type="submit"` and have the submit hander on the form elements `onSubmit` attribute and not on the button itself. This is more semantically correct (aka better for accessibility) as we are using this button as a submit button for the form. There could be other buttons in the form that are not submit buttons and those can default to `type="button"`.

Reply:
Sounds good. Confirmed that this works in the case of this PR, and agree, as all FE engs did at
sync, that the semantic wins are worth it. Thanks for calling this out.